### PR TITLE
UI block type examples

### PR DESCRIPTION
**Pull request by AaronVanVyve**
Improved UIBlockTypes examples with functionality and explanations

### CODE REVIEW

**Comment by MariekeGO** [comment]
Could you correct these unresolved cross-references?
image

File: develop/devguide/Automation/UIBlockTypesOverview.md [hunk 1]
```diff
@@ -31,22 +52,41 @@ Allows you to define a newly created dialog box item as a button.
...
 ```csharp
-UIBlockDefinition blockItem = new UIBlockDefinition();
-blockItem.Type = UIBlockType.Calendar;
-...
+UIBlockDefinition blockItem = new UIBlockDefinition
+{
+  Type = UIBlockType.Calendar,
+  InitialValue = DateTime.Now.ToString("dd/MM/yyyy HH:mm:ss"),
+  DestVar = "InputCalendar",
+  Row = 1,
```
Comment (Skyline-ThomasGH) [inline develop/devguide/Automation/UIBlockTypesOverview.md:87]: We could remove these row & column lines in all examples to safe some lines. These are generic fields that (should) behave the same for all blocks. The same could be said for the DestVar tbh, maybe only fill it in when the  block definition is accompanied by checking logic?

**Review by Skyline-ThomasGH — CHANGES_REQUESTED** [review]
These changed are a great improvement over the existing state of this page. There is still a lot of missing information, but this is a good step in the right direction. We should discuss whether it is worth the effort of providing a fully detailed doc page for this in an upcoming retro.

File: develop/devguide/Automation/UIBlockTypesOverview.md [hunk 2]
```diff
@@ -310,55 +469,36 @@ To retrieve the results:
...
-     next.Column = 0;
-     next.Text = "Next";
-     next.DestVar = "Next";
-     uib.AppendBlock(next);
-    
-     _treeResults = _engine.ShowUI(uib);
-} while (!_treeResults.WasButtonPressed("Next"));
-
-ShowResult();
+  Type = UIBlockType.TreeView,
```
Comment (Skyline-ThomasGH) [inline develop/devguide/Automation/UIBlockTypesOverview.md:474]: Maybe mention the 'using Skyline.DataMiner.Net.AutomationUI.Objects;' statement wich is required for the TreeViewItem.

File: develop/devguide/Automation/UIBlockTypesOverview.md [hunk 3]
```diff
@@ -6,23 +6,44 @@ uid: UIBlockTypesOverview
...
+| [TextBox](#textbox) | Text box. |
+| [Time](#time) | Item that displays a time value. |
+| [TreeView](#treeview) | Tree view control. |
+
+## UIBuilder
+
+To create these UIBlocks, a UIBuilder should be defined with *Width*, *RowDefs*, and *ColumnDefs*.
+
+```csharp
+var MyDialogBox = new UIBuilder() { Width = 800, RowDefs = "auto;auto", ColumnDefs = "auto;auto" };
```
Comment (Skyline-ThomasGH) [inline develop/devguide/Automation/UIBlockTypesOverview.md:32]: Would change 'MyDialogBox' to 'uiBuilder' and make sure it is referenced everywhere like this. I found at least one (Numeric section) where something else is used.

**Review by Skyline-ThomasGH — COMMENTED** [review]
Two additional notes.

**Review by Skyline-ThomasGH — APPROVED** [review]

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**